crates/subspace-farmer-components/src/auditing.rs

```diff
@@ -165,7 +178,18 @@ where
                 best_solution_distance,
             })
         })
-        .collect()
+        .collect();
+
+    if let Some(audit_event_handler) = audit_event_handler {
+        let duration = Instant::now().duration_since(start);
+        let duration_secs = duration.as_secs_f64();
+
+        audit_event_handler.call_simple(&AuditEvent {
+            duration: duration_secs,
+        });
+    }
```

Comment:
Since this is happening at the end of the `audit_plot_sync` call, you might as well surface it to `subspace-farmer` and not change `subspace-farmer-components` at all. Ideally we wouldn't need to change anything in this crate.

Reply:
The proposed change will also include additional time for PoT generation. Please, confirm that you want to measure the `audit()` call timing instead.

Reply:
PoT generation is Timekeeper's responsibility, can you clarify what you mean here?

Reply:
I meant this line: https://github.com/subspace/subspace/blob/f76fb316a42728ff15dc39b00d6475fb910220f7/crates/subspace-farmer/src/single_disk_farm/farming.rs#L170

Should I measure the whole `audit()` method or just `audit_plot_sync()` invokation within it?

Reply:
That is PoSpace table. And no, I don't think that is necessary to measure it, that is too detailed.

Reply:
BTW that callback with table generator is only called when proving is done. I guess it might have been confusing to see it in this file, but it is not actually running just yet.